### PR TITLE
PPP: Featured image not displayed

### LINKED ISSUE #11188: [Properly Placed Pages] Pages list: featured image is not loaded
### Expected behavior
Opening the pages list I should be able to see a featured image if a page has one.

### Actual behavior
simulator screen shot - iphone x - 2019-03-05 at 16 08 17

### PR DESCRIPTION
**Pull request by danielebogo**
Fixes #11188 

This PR fixes a problem with the Page and its featured image. At the moment, if a Page featured image is set up from outside (using the website or another device), is not possible to see that image when a blog is synchronised. This because if the featured image is not present in the local data model a new model is created marked as _stub_ containing only the media id.
Also the `ImageLoader` used to load the featured image returns if a _Media_ doesn't contain a valid URL.

**To test:**
- Select a blog using another device
- Open the pages list and set a featured image
- Select the same blog on your device and open the pages list
- You should all the featured images

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Pages/PageListTableViewCell.m [hunk 1]
```diff
@@ -177,11 +166,20 @@ - (void)configureFeaturedImage
...
-                                       preferredSize:CGSizeMake(FeaturedImageSize, FeaturedImageSize)
-                                         placeholder:nil
-                                             success:nil
-                                               error:nil];
+        [self.featuredImageView startLoadingAnimation];
+
+        __weak __typeof(self) weakSelf = self;
+
+        [page.featuredImage imageWithSize:CGSizeMake(FeaturedImageSize, FeaturedImageSize)
+                        completionHandler:^(UIImage * _Nullable result, NSError * _Nullable error) {
```
Comment (aerych) [inline WordPress/Classes/ViewRelated/Pages/PageListTableViewCell.m]: I'm wondering about a case where a cell has started loading its featured image, but is then reused before the image finishes loading and starts loading a different featured image.  Depending on network latency I think we could potentially see the second image be fetched first, and then be replaced when the first image finished loading. (Does that make sense?)

Maybe we could add a check in the completion block that the cell's current page is the same as the page that had the newly acquired featured image and only set the image it if that is true?

File: WordPress/Classes/ViewRelated/Pages/PageListTableViewCell.m [hunk 2]
```diff
@@ -177,11 +166,20 @@ - (void)configureFeaturedImage
...
+                            __strong __typeof(weakSelf) strongSelf = weakSelf;
+                            [strongSelf.featuredImageView stopLoadingAnimation];
+                            
+                            if (error == nil) {
```
Comment (aerych) [inline WordPress/Classes/ViewRelated/Pages/PageListTableViewCell.m]: I might have missed it, but I didn't see where we were logging the error anywhere.  Maybe we could add call to DDLogError when there is an error to help with any future debugging should there be a need?

**Comment by danielebogo** [comment]
At the end i came with another solution. First thing was to modify a bit the `MediaThumbnailCoordinator` adding a new method to fetch remotely a Media from a stub media.
Then I was able to update the `loadImage(media:, preferredSize:, placeholder:, success:, error:)` method (thanks @etoledom for the _reusability_ clarification 😊) without duplicating the method.
@aerych the PR is ready for another look
@etoledom do you have a minute to have a look as well?

**Comment by loremattei** [comment]
I'm moving this PR to 12.1 since 12.0 has been cut. Please, if you want it to make it to 12.0, feel free to move it back, target the release branch and ping me to build a new beta.

**Comment by danielebogo** [comment]
Thanks @loremattei , 12.1 works well too!

**Comment by danielebogo** [comment]
Thanks @aerych

**Comment by danielebogo** [comment]
Thanks @etoledom

**Pull request merged**